SQL/TSQL.sublime-syntax

```diff
@@ -0,0 +1,841 @@
+    - match: \b(?i:open|fetch(?:(?:\s+next)?\s+from)?|close|deallocate)\b
+      scope: keyword.other.sql
+      push:
+        - cursor-name
+        - single-identifier-after-whitespace
+    - match: \b(?i:merge)\b
+      scope: keyword.other.tsql
+      push:
+        - maybe-table-alias
+        - table-name
+        - single-identifier-after-whitespace
+    - match: \b(?i:using)\b
+      scope: keyword.other.tsql
+      push:
+        - table-name-or-subquery
+        - join-on
+        - maybe-table-alias
+        - table-name
+        - single-identifier-after-whitespace
+    - match: \b(?i:set)\b
+      scope: keyword.other.dml.tsql
+
+  ddl-statements:
+    - meta_prepend: true
+    - match: (?i)\b(?:create(?:\s+or\s+alter)?)\b(?!\s*table\b)
+      scope: keyword.other.ddl.sql
+      push:
+        - ddl-create-target-expect-as
+        - ddl-create-target
+        - create-table-condition
+        - ddl-target
+
+  ddl-create-target-expect-as:
+    - meta_prepend: true
+    - include: with
+
+  ddl-target:
+    - meta_prepend: true
+    - match: \b(?i:proc|unique|clustered|nonclustered)\b
+      scope: keyword.other.ddl.sql
+
+  ddl-alter-target:
+    - meta_prepend: true
+    #- match: \b(?i:proc|procedure|function)\b
+    #  scope: keyword.other.sql
+    - match: (?=\S)
+      set:
+        - ddl-create-target-expect-as
+        - expect-procedure-name
+
+  top:
+    - match: (?i)\b(top)\b(?:\s*(?:(\()\s*)?(\d+)(?:\s*(\)))?(?:\s+(percent\b))?)?
+      captures:
+        1: keyword.other.dml.tsql
+        2: meta.group.tsql punctuation.section.parens.begin.tsql
+        3: meta.group.tsql meta.number.integer.decimal.tsql constant.numeric.value.tsql
+        4: meta.group.tsql punctuation.section.parens.end.tsql
+        5: keyword.other.dml.tsql
+
+  dml-delete:
+    - meta_prepend: true
+    - include: top
+
+  dml-update:
+    - meta_prepend: true
+    - include: top
+
+  built-in-scalar-function-calls:
+    - meta_append: true
+    - match: (?i)\b(CONVERT)\s*(\()
+      scope: meta.function-call.sql
+      captures:
+         1: support.function.scalar.sql
+         2: meta.group.sql punctuation.section.arguments.begin.sql
+      push:
+        - method-call-convert
+        - expect-type
+    - match: (?i)\b(?:GETDATE)(?=\s*\()
+      scope: support.function.scalar.sql
+      push: begin-method-call-paren
+
+  method-call-convert:
+    - meta_content_scope: meta.function-call.sql meta.group.sql
+    - match: \)
+      scope: meta.function-call.sql meta.group.sql punctuation.section.arguments.end.sql
+      pop: 1
+    - match: ','
+      scope: punctuation.separator.argument.sql
+    - include: expressions-or-column-name
+
+  label-name:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.label-name.sql
+    - match: ''
+      pop: 1
+
+  cursor-name:
+    - meta_content_scope: meta.cursor-name.sql
+    - match: ''
+      pop: 1
+
+  after-declare:
+    - match: (?=\w+\s+(?i:cursor)\b)
```

Comment:
Are cursor names suppost to be simple unquoted identifiers only? `\w+` restricts it this way, while `single-identifer` indicates the opposite.

Reply:
good question - I had only ever seen unquoted identifiers used, but it seems like any identifier is possible. I guess I'll have to fix this. From https://docs.microsoft.com/en-us/sql/t-sql/language-elements/declare-cursor-transact-sql?view=sql-server-ver16#arguments:

> cursor_name
Is the name of the Transact-SQL server cursor defined. cursor_name must conform to the rules for identifiers.

Reply:
actually after testing on SQL Fiddle, it seems that it refers to unquoted identifiers only

Reply:
So the whole pattern could be simplified to directly consume the identifer rather than pushing into simple-identifier followed by meta context.